internal/command/resourceplugin/subst.go

```diff
@@ -17,13 +19,27 @@ func loadResourcePlugin(file string, tplVals config.TemplateVals, forDelete bool
 }
 
 func unstructuredToResourcePlugin(un any) (*models.ResourcePlugin, error) {
-	d, err := json.Marshal(un)
+	var (
+		name string
+		ok   bool
+		spec any
+	)
+	switch x := un.(type) {
+	case map[string]any:
+		name, ok = x["name"].(string)
+		spec = x["spec"]
+	default:
+	}
+	if !ok {
+		return nil, errors.New("missing name or spec fields")
+	}
+	d, err := json.Marshal(spec)
```

Comment:
Can we place this logic into a function? (and call it from `resourceplugin`, `routegroup` and `sandbox`)

Reply:
sure, done.